src/web/components/ApiKeyManagement/KeyEditDialog.tsx

```diff
@@ -0,0 +1,79 @@
+import { useState } from 'react';
+import { SubmitHandler } from 'react-hook-form';
+
+import { ApiRoleDTO } from '../../../api/entities/ApiRole';
+import { ApiKeyDTO } from '../../../api/services/adminServiceHelpers';
+import { EditApiKeyFormDTO } from '../../services/apiKeyService';
+import { Dialog } from '../Core/Dialog';
+import { Form } from '../Core/Form';
+import { CheckboxInput } from '../Input/CheckboxInput';
+import { TextInput } from '../Input/TextInput';
+
+export type OnApiKeyEdit = (
+  form: EditApiKeyFormDTO,
+  setApiKey: React.Dispatch<React.SetStateAction<ApiKeyDTO>>,
+  participantId?: number
+) => void;
+
+type KeyEditDialogProps = {
+  onEdit: OnApiKeyEdit;
+  triggerButton: JSX.Element;
+  availableRoles: ApiRoleDTO[];
+  apiKey: ApiKeyDTO;
+  setApiKey: React.Dispatch<React.SetStateAction<ApiKeyDTO>>;
+};
+
+function KeyEditDialog({
+  availableRoles,
+  onEdit,
+  triggerButton,
+  apiKey,
+  setApiKey,
+}: KeyEditDialogProps) {
+  const [open, setOpen] = useState(false);
+
+  const onSubmit: SubmitHandler<EditApiKeyFormDTO> = async (formData) => {
+    await onEdit(formData, setApiKey);
+    setOpen(false);
+  };
+
+  const possibleRolesMap = new Map<number, ApiRoleDTO>();
```

Comment:
I wonder if we could extract some of this away, so that it just becomes 
```
  const allowedRoles = getAllowedRoles(availableRoles, apiKey.roles);
```
which is an array so that you don't need the 
```
Array.from(possibleRolesMap...
```
in the `options` of the `CheckboxInput` below.

Reply:
Yeah sounds good